src/main/java/io/rpg/view/popups/DialoguePopup.java

```diff
@@ -0,0 +1,81 @@
+package io.rpg.view.popups;
+
+import io.rpg.viewmodel.DialoguePopupViewModel;
+import javafx.event.EventHandler;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Group;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.image.Image;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.paint.Color;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public class DialoguePopup extends Scene {
+
+  private final DialoguePopupViewModel viewModel;
+  private final List<String> textPages;
+  private int currentPage = 0;
+
+  public DialoguePopup(String text, Image npcImage, String backgroundPath, String buttonPath) {
+    this(text, npcImage);
+    viewModel.setBackgroundImage(backgroundPath);
+    viewModel.setCloseButtonImage(buttonPath);
+  }
+
+  public DialoguePopup(String text, Image npcImage) {
+    super(new Group(), Color.TRANSPARENT);
+    FXMLLoader loader = new FXMLLoader(Objects.requireNonNull(DialoguePopupViewModel.class.getResource("dialogue-popup-view.fxml")));
+    Parent root = null;
+
+    try {
+      root = loader.load();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    this.setRoot(root);
+
+    viewModel = loader.getController();
+    viewModel.setNpcImage(npcImage);
+    viewModel.setBackgroundImage("file:assets/popup-background.png"); //images do not load from fxml on my computer
```

Comment:
might be a little problem with working with this code, maybe it's because of operating system, but it's fine as long as it works on any of our computers, the app doesn't need to be cross-platform